services/search/pkg/engine/bleve.go

```diff
@@ -370,7 +370,22 @@ func unmarshalInterfaceMap(out any, flatMap map[string]interface{}, prefix strin
 		if value, ok := flatMap[mapKey]; ok {
 			if field.Kind() == reflect.Ptr {
 				alloc := reflect.New(field.Type().Elem())
-				alloc.Elem().Set(reflect.ValueOf(value).Convert(field.Type().Elem()))
+				elemType := field.Type().Elem()
+
+				// Handle *time.Time specifically
```

Comment:
nitpick: this isn't exactly correct anymore 
I would just get rid of the comment